File: android/SherpaOnnxJavaDemo/app/src/main/java/com/k2fsa/sherpa/onnx/MainActivity.java
package com.k2fsa.sherpa.onnx;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;
import androidx.lifecycle.ViewModelProvider;

import android.Manifest;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import com.k2fsa.sherpa.onnx.service.SpeechSherpaRecognitionService;

import pub.devrel.easypermissions.EasyPermissions;

public class MainActivity extends AppCompatActivity {
    private AppViewModel appViewModel;
    private TextView tvText;
    private static final int RC_AUDIO_PERM = 123;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        tvText = findViewById(R.id.text);
        requestMicrophonePermission();
    }


    private void startSpeechService() {
        Intent serviceIntent = new Intent(this, SpeechSherpaRecognitionService.class);
        ContextCompat.startForegroundService(this, serviceIntent);
        appViewModel = new ViewModelProvider(Application.getInstance()).get(AppViewModel.class);
        appViewModel.getSpeechRecognitionResult().observe(this, this::handleSpeechRecognitionResult);
    }

    private void handleSpeechRecognitionResult(String result) {
        tvText.setText(result);
    }

    private void requestMicrophonePermission() {
        String[] perms = {Manifest.permission.RECORD_AUDIO};
        if (EasyPermissions.hasPermissions(this, perms)) {
            startSpeechService();
        } else {
            EasyPermissions.requestPermissions(MainActivity.this,
                    "We need access to your microphone for voice recognition",
                    RC_AUDIO_PERM, perms);
        }
    }
}

File: android/SherpaOnnxVad/app/src/main/java/com/k2fsa/sherpa/onnx/MainActivity.kt
package com.k2fsa.sherpa.onnx

import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {

    private lateinit var recordButton: Button
    private lateinit var circle: View

    @Volatile
    private var isRecording: Boolean = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        circle= findViewById(R.id.powerCircle)

        recordButton = findViewById(R.id.record_button)
        recordButton.setOnClickListener { onclick() }
    }

    private fun onclick() {
        if (!isRecording) {
            isRecording = true
            onVad(true)
        } else {
            isRecording= false
            onVad(false)
        }
    }

    private fun onVad(isSpeech: Boolean) {
        if(isSpeech) {
            circle.background = resources.getDrawable(R.drawable.green_circle)
        } else {
            circle.background = resources.getDrawable(R.drawable.gray_circle)
        }

    }
}